src/wallet/rpcwallet.cpp

```diff
@@ -3869,16 +3869,16 @@ UniValue z_getbalance(const UniValue& params, bool fHelp)
             nBalance = getBalanceZaddr(addr, std::nullopt, nMinDepth, INT_MAX, false);
         },
         [&](const libzcash::UnifiedAddress& addr) {
-            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true, false);
+            auto selector = pwalletMain->ZTXOSelectorForAddress(addr, true, TransparentCoinbasePolicy::Allow, TransactionStrategy(PrivacyPolicy::FullPrivacy));
```

Comment:
This seems like it should use `NoPrivacy`, so that for balance purposes we get everything. This points again to the fact that requiring the transaction strategy here is just a little werid.

Reply:
Specifying `FullPrivacy` here means that we will exclusively use address-based selection, and not account-based selection, when querying for the balance of a UA; really, we need to get https://github.com/zcash/zcash/pull/6389 in and deprecate the UA support here.

Reply:
Yeah, `FullPrivacy` is equivalent to `false` here. But you’re right – `TransactionStrategy` is the wrong thing. I’ll revert this back to `bool` for now, for a couple reasons:
1. `TransparentCoinbasePolicy` avoids the bug (caused by too many bools) that led to this change in the first place and
2. I have a better approach that uses accounts more broadly (#6457 – only the latest commit), and gives a better name to the parameter that separates it from the idea of privacy policies (although those are still used to determine the argument in some cases).